binr/ragg2/ragg2.c

```diff
@@ -108,6 +109,257 @@ static int openfile (const char *f, int x) {
 }
 #define ISEXEC (fmt!='r')
 
+static char* parseCFile(const char *file, const char *arch, const char *os, int bits) {
+	char *SFLIBPATH = NULL;
+	char CC[100] = "";
+	char OBJCOPY[10] = "";
+	char CFLAGS[200] = "";
+	char LDFLAGS[200] = "";
+	char buffer[200] = "";
+	char JMP[4] = "";
+	char FMT[6] = "";
+	char SHDR[100] = "";
+	char TRIPLET[30] = "";
+	char TEXT[20] = "";
+	bool use_clang;
+	char *output = NULL;
+	int fileExtFd = 0; 
+	char *fileExt = NULL; // "file" with extension (.s, .text, ...)
+
+	// Find compiler
+	output = r_sys_getenv("CC");
+	if (!output || (strcmp (output, "llvm-gcc") 
+  		&& strcmp (output, "clang") && strcmp (output, "gcc")))
+	{
+	   	output = r_sys_cmd_strf ("which llvm-gcc");
+		if (output) {
+			strcpy (CC, "llvm-gcc");
+		}
+
+		free (output);
+		output = r_sys_cmd_strf ("which clang");
+		if (output) {
+			strcpy (CC, "clang");
+		}
+
+		free (output);
+		output = r_sys_cmd_strf ("which gcc");
+		if (output) {
+			strcpy (CC, "gcc");
+		}
+
+		if (CC[0] == '\0') {
+			eprintf ("Please set CC before running.\n");
+			goto fail;
+		}
+	} else {
+		strncpy (CC, output, sizeof (CC) - 1);
+	}
+
+	// Get path for sflib
+	SFLIBPATH = r_sys_getenv("SFLIBPATH");
+	if (!SFLIBPATH) {
+		free (output);
+		output = r_sys_cmd_strf ("r2 -hh | grep INCDIR | awk '{print $2}'");
+		if (!strcmp (output, "\0")) {
+			eprintf ("Cannot find SFLIBPATH env var.\n"
+		  		 "Please define it, or fix r2 installation.\n");
+			goto fail;
+		}
+
+		output[strcspn (output, "\n")] = '\0';
+		if ((SFLIBPATH = malloc (strlen (output) + 7)) == NULL) {
+			eprint ("Malloc for sflibpath failed!\n");
+			goto fail;
+		}
+		sprintf (SFLIBPATH, "%s/sflib", output);
+	}
+
+	strcpy (JMP, "jmp");
+	if (!strcmp (arch, "arm") || !strcmp (arch, "arm64") || !strcmp (arch, "aarch64")
+	  	|| !strcmp (arch, "thumb") || !strcmp (arch, "arm32") || !strcmp (arch, "mips")
+		|| !strcmp (arch, "mips32") || !strcmp (arch, "mips64")) {
+		strcpy (JMP, "b");
+	}
+
+	strcpy (FMT, "elf");
```

Comment:
Better put all those vars into a struct and move this into a function that fills the fields